adb.js

```diff
@@ -0,0 +1,162 @@
+import usb from 'usb';
+import { USB_VENDOR_IDS, ADB_VALUES, CONNECTION_TYPES } from './lib/constants';
+import { logExceptOnTest } from './lib/helpers';
+import ADBDevice from './lib/adb-device';
+
+const NOT_CONNECTED = 0;
+const WAIT_FOR_AUTH = 1;
+const SEND_PRIVATE_KEY = 2;
+const SEND_PUBLIC_KEY = 3;
+const CONNECTED = 4;
+
+class ADB {
+  constructor (connectionType, device) {
+    if (connectionType !== (CONNECTION_TYPES.USB || CONNECTION_TYPES.TCP)) {
+      throw new Error("Cannot create new ADB device, invalid connection type.");
+    }
+    this.state = NOT_CONNECTED;
+    this.device = new ADBDevice(connectionType, device);
+  }
+
+  // *** STATIC FUNCTIONS ***
+  // return an array of devices that have an ADB interface
+  static findAdbDevices () {
+    logExceptOnTest("Trying to find a usb device");
+    let adbDevices = [];
+    let usbDevices = usb.getDeviceList();
+    let deviceIndex = 0;
+    for (let device of usbDevices) {
+      let deviceInterface = this.getAdbInterface(device);
+      if (deviceInterface !== null) {
+        logExceptOnTest("Found an ADB device");
+        adbDevices.push({device, deviceInterface});
+        deviceIndex++;
+      }
+    }
+    return adbDevices;
+  }
+
+  // search through a devices interfaces for an interface
+  // that can be used for ADB communications
+  static getAdbInterface (device) {
```

Comment:
This method will return one of three values: `null`, `undefined`, or the adb interface. It would be better, for consistency's sake, to limit to two (either initializing `returnInterface` to `undefined`, or returning `null` at the end).